### PR TITLE
Add tests for scaled_mm + DTensor

### PR DESCRIPTION
**Pull request by wanchaol**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #210
* __->__ #209

This PR adds test to test the scaled_mm op working with DTensor.

Currently using the same style of test runner
compared to other tests in float8_experimental. We shall consider
switching to DTensorTestBase or MultiProcessTestCase in the repo but
didn't do it in this PR

Differential Revision: [D53731879](https://our.internmc.facebook.com/intern/diff/D53731879)

### CODE REVIEW

File: test/test_dtensor.py [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+Test numerics of manually defined float16 TP vs float8 TP of toy models
+"""
+
+import copy
+import os
+
+import torch
+import torch.nn as nn
+
+# from fairscale.nn.model_parallel.layers import ColumnParallelLinear, RowParallelLinear
```
Comment (drisspg) [inline test/test_dtensor.py:16]: was this for comparison checks?

**Review by drisspg — APPROVED** [review]
Awesome, thank you for adding this!

File: test/test_dtensor.py (hunk 1 above)
Comment (wanchaol) [reply test/test_dtensor.py:16]: good catch! I should remove it actually

**Comment by wanchaol** [comment]
@wanchaol has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D53731879).

**Comment by wanchaol** [comment]
@wanchaol has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D53731879).

**Pull request closed without merging**